code/src/layouts/header/Header.test.tsx

```diff
@@ -0,0 +1,23 @@
+import { MemoryRouter } from "react-router-dom";
+import { screen, render } from "@testing-library/react";
+import Header from "@/layouts/header/Header";
+
+// @TODO: remove or move this mock
+global.IntersectionObserver = jest.fn(() => ({
+  observe: jest.fn(),
+  unobserve: jest.fn(),
+  disconnect: jest.fn(),
+  takeRecords: jest.fn(),
+  root: null,
+  rootMargin: "0px",
+  thresholds: []
+}));
+
+test("renders Header correctly", () => {
+  render(
+    <MemoryRouter>
+      <Header />
+    </MemoryRouter>
+  );
+  expect(screen.getByText("login.label")).toBeInTheDocument();
```

Comment:
move to separate variable screen.getByText("login.label")

Reply:
Fixed